### ISSUE TITLE
Places365 dataset redundant path.join under _check_integrity

### ISSUE THREAD

**Issue by YellowPig-zp** [pytorch/vision#3501]
## 🐛 Bug

Redundant path.join at line 165 under _check_integrity in torchvision/datasets/places365.py

path.join is already called at line 116 and line 131 under load_categories and load_file_list

cc @pmeier

**Comment by pmeier** [comment]
While I at first was under the impression that this is a serious bug, it turns out @YellowPig-zp was right and it is just redundant. The documentation of `os.path.join` [states](https://docs.python.org/3/library/os.path.html#os.path.join):

> If a component is an absolute path, all previous components are thrown away and joining continues from the absolute path component.

Example:

```python
import os

root = os.path.join(os.sep, "root")
filename = "file"
file = os.path.join(root, filename)


assert os.path.join(root, file) == file

```

Since `file` is an absolute path after

https://github.com/pytorch/vision/blob/7992eb5da9c2e67469734e43f3d07e242d4f5273/torchvision/datasets/places365.py#L115-L116

or

https://github.com/pytorch/vision/blob/7992eb5da9c2e67469734e43f3d07e242d4f5273/torchvision/datasets/places365.py#L130-L131

the `path.join` in 

https://github.com/pytorch/vision/blob/7992eb5da9c2e67469734e43f3d07e242d4f5273/torchvision/datasets/places365.py#L165

is equal to `file`.

---

**Comment by YellowPig-zp** [comment]
Thanks for fixing this! Just want to point out that I first encountered the issue when I used **relative path** rather than **absolute path**, and unfortunately it would cause an error if relative path is used.

For example if I want to download the dataset to the folder my_data/places365 under the current working directory I would call 
`data = Places365(root='./my_data/places365', split='train-standard', download=True)`
However, once it's downloaded and if I want to use the dataset again, I would call
`data = Places365(root='./my_data/places365', split='train-standard', download=False)` with the download flag marked False.

This would throw me an error saying the path to some files doesn't exist. Removing the path.join in check_integrity would quickly fix the bug, as did in #3545 .
